pkg/network/http/monitor.go

```diff
@@ -29,20 +28,11 @@ type Monitor struct {
 	telemetry     *telemetry
 	pollRequests  chan chan struct{}
 	closeFilterFn func()
+	statkeeper    *httpStatKeeper
 }
 
 // NewMonitor returns a new Monitor instance
-func NewMonitor(procRoot string, m *manager.Manager, h *ddebpf.PerfHandler) (*Monitor, error) {
-	filter, _ := m.GetProbe(manager.ProbeIdentificationPair{Section: string(probes.SocketHTTPFilter)})
-	if filter == nil {
-		return nil, fmt.Errorf("error retrieving socket filter")
-	}
-
-	closeFilterFn, err := network.HeadlessSocketFilter(procRoot, filter)
-	if err != nil {
-		return nil, fmt.Errorf("error enabling HTTP traffic inspection: %s", err)
-	}
-
```

Comment:
Curious, why did you change this?

Reply:
It was because there was an import cycle between the network & http packages - but this doesn't fix the tests, so I removed this change and instead made a new `filter` package to put `HeadlessSocketFilter` in